### PR TITLE
drivers/ads101x : Expose Configurations to Kconfig 

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in ADS101X sensor driver to Kconfig.

### Testing procedure

The firmware was uploaded to FIT/IoT-LAB Test Bed and following results were obtained.

##### Firmware Output
main(): This is RIOT! (Version: 2020.07-devel-55-g1a2b1-kconfig_ads101x)
CONFIG_ADS101X_I2C_ADDRESS=(0x48)
ADS101X analog to digital driver test application

Initializing ADS101x analog to digital at I2C_DEV(0)... [Failed]

#### Usage with CFLAGS 

/tests/driver_ads101x/Makefile

> CFLAGS += -DCONFIG_ADS101X_I2C_ADDRESS=0x49

##### Firmware Output
main(): This is RIOT! (Version: 2020.07-devel-55-g1a2b1-kconfig_ads101x)
CONFIG_ADS101X_I2C_ADDRESS=0x49
ADS101X analog to digital driver test application

Initializing ADS101x analog to digital at I2C_DEV(0)... [Failed]

#### Usage with Kconfig

/tests/ads101x/

> make menuconfig

##### Firmware Output
main(): This is RIOT! (Version: 2020.07-devel-55-g1a2b1-kconfig_ads101x)
CONFIG_ADS101X_I2C_ADDRESS=0x51
ADS101X analog to digital driver test application

Initializing ADS101x analog to digital at I2C_DEV(0)... [Failed]

Note : The sensor is not available hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
Please rebase

**Comment by akshaim** [comment]
> Please rebase

Done.

File: drivers/Kconfig.net [hunk 1]
```diff
@@ -11,4 +11,5 @@ endmenu # Network Device Drivers
 
 menu "Sensor Device Drivers"
 rsource "hdc1000/Kconfig"
-endmenu # Sensor Device Drivers
+rsource "ads101x/Kconfig"
```
Comment (leandrolanzieri) [inline drivers/Kconfig.net:14]: Please keep alphabetical order.

File: drivers/ads101x/Kconfig [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+menuconfig KCONFIG_MODULE_ADS101X
+    bool "Configure ADS101X driver"
+    depends on MODULE_ADS101X
+    help
+        Configure the ADS101X driver using Kconfig.
+
+if KCONFIG_MODULE_ADS101X
+
+config ADS101X_I2C_ADDRESS
+    hex "I2C Default ADDRESS"
```
Comment (leandrolanzieri) [inline drivers/ads101x/Kconfig:16]:
```suggestion
    hex "Default I2C address"
```

File: drivers/ads101x/Kconfig [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    range 0x48 0x51
+    default 0x48
+    help
+        ADS101X allows for upto 4 devices on Single Bus.
+        The value depends on the state of ADDR Pin.
+        Default value (0x48) corresponds to ADDR pin tied to GND.
+        For more information refer I2C Address Selection in Datasheet.
```
Comment (leandrolanzieri) [inline drivers/ads101x/Kconfig:23]:
```suggestion
        ADS101X allows for upto 4 devices on Single Bus. The value depends on
        the state of ADDR Pin. Default value (0x48) corresponds to ADDR pin tied
        to GND. For more information refer I2C Address Selection in Datasheet.
```

File: drivers/Kconfig.net [hunk 4]
```diff
@@ -10,5 +10,6 @@ source "$(RIOTCPU)/nrf52/radio/nrf802154/Kconfig"
 endmenu # Network Device Drivers
 
 menu "Sensor Device Drivers"
+rsource "ads101x/Kconfig"
 rsource "hdc1000/Kconfig"
-endmenu # Sensor Device Drivers
+endmenu # Sensor Device Driver
```
Comment (leandrolanzieri) [inline drivers/Kconfig.net:15]: Unrelated change. Please remove

Comment (akshaim) [reply drivers/Kconfig.net]: Done. removed and rebased.
